### PR TITLE
Create Java Solution for Milk Factory

### PR DESCRIPTION
**Pull request by MrinallU**

### CODE REVIEW

**Comment by MrinallU** [comment]
@bqi343 I think this should be good to merge.

File: solutions/USACO open19-milk-factory.mdx
Comment (frodakcin) [inline solutions/USACO open19-milk-factory.mdx:82]:
```suggestion
            if(ans == N) {
```

Comment (frodakcin) [reply solutions/USACO open19-milk-factory.mdx:82]: For clarity

File: solutions/USACO open19-milk-factory.mdx
Comment (frodakcin) [inline solutions/USACO open19-milk-factory.mdx:68]: This array is being passed by reference, right? (as opposed to copy)

**Review by frodakcin — CHANGES_REQUESTED** [review]
Mostly good. Just a slight change I'd like

**Comment by MrinallU** [comment]
@frodakcin Alright that is fixed! Thanks for pointing that one out.

File: solutions/USACO open19-milk-factory.mdx
Comment (MrinallU) [reply solutions/USACO open19-milk-factory.mdx:68]: @frodakcin I believe it is.

**Review by frodakcin — APPROVED** [review]
Gonna wait for a couple more reviews

**Review by dongliu0426 — APPROVED** [review]

**Review by Farrius — APPROVED** [review]

File: solutions/USACO open19-milk-factory.mdx
Comment (bqi343) [inline solutions/USACO open19-milk-factory.mdx:88]: bugs me that you don't seem to be consistent about spacing D:

`for (int i = 0; i < N; i++) {`

`if(ans == N) {`

`if(!used){`

```
while (i.hasNext())
        {
```

Why is there so much space in the `addEdge` function?

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO open19-milk-factory.mdx
Comment (darren-yao) [reply solutions/USACO open19-milk-factory.mdx:88]: @MrinallU I'll approve once you fix formatting and spacing consistency.

**Comment by MrinallU** [comment]
@bqi343 The spacing should be consistent now. I use Vm so I haven't found a good formatting plugin yet, so I apologize for the formatting inconsistencies.

File: solutions/USACO open19-milk-factory.mdx
Comment (bqi343) [inline solutions/USACO open19-milk-factory.mdx:82]: spacing for these if statements are different from the one above

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Comment by MrinallU** [comment]
@bqi343 How about now?

File: solutions/USACO open19-milk-factory.mdx
Comment (bqi343) [inline solutions/USACO open19-milk-factory.mdx:30]: now this for loop is different :(

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO open19-milk-factory.mdx
Comment (MrinallU) [reply solutions/USACO open19-milk-factory.mdx]: Oh, I should be good now

File: solutions/USACO open19-milk-factory.mdx
Comment (dongliu0426) [inline solutions/USACO open19-milk-factory.mdx:40]:
```suggestion
         int ans = 0;
```

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

**Review by darren-yao — APPROVED** [review]
ok

**Pull request merged**
